engine/docs-runner/src/main/scala/org/enso/docsrunner/Main.scala

```diff
@@ -0,0 +1,50 @@
+package org.enso.docsrunner
+
+import java.io.File
+import org.enso.interpreter.runtime.{Context => EnsoContext}
+import org.enso.languageserver.io.{
+  ObservableOutputStream,
+  ObservablePipedInputStream
+}
+import org.enso.polyglot.{LanguageInfo, MethodNames}
+import org.graalvm.polyglot.Context
+
+object Main {
+  val stdOut    = new ObservableOutputStream
+  val stdErr    = new ObservableOutputStream
+  val stdInSink = new ObservableOutputStream
+  val stdIn     = new ObservablePipedInputStream(stdInSink)
+
+  def generateFrom(file: File): Unit = {
+    val executionContext = Context
+      .newBuilder()
+      .allowAllAccess(true)
+      .allowExperimentalOptions(true)
+      .out(stdOut)
+      .err(stdErr)
+      .in(stdIn)
+      .build()
+
+    val languageContext = executionContext
+      .getBindings(LanguageInfo.ID)
+      .invokeMember(MethodNames.TopScope.LEAK_CONTEXT)
+      .asHostObject[EnsoContext]
+
+    val module = languageContext.getModuleForFile(file)
+
+    val generated = module.map(languageContext.getCompiler.generateDocs)
```

Comment:
This should be a polyglot method on a module. Look at this: https://github.com/enso-org/enso/blob/main/engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java#L407 – you'll have to add your own method name constant close to the other constants and your own case branch. Notice that `Context` is already argument to the `doInvoke` method, so you don't have to leak it from here.

Then add the polyglot-wrapper for this method here: https://github.com/enso-org/enso/blob/main/engine/polyglot-api/src/main/scala/org/enso/polyglot/Module.scala